src/base/static/components/place-detail/index.js

```diff
@@ -210,6 +215,90 @@ class PlaceDetail extends Component {
     });
   }
 
+  onAdditionalData(action, payload) {
+    switch (action) {
+      case constants.ON_ADD_ATTACHMENT_ACTION:
+        this.attachments.push(payload);
+        break;
+      default:
+        console.error(
+          "Error: Unable to handle form field callback action:",
+          action
+        );
+        break;
+    }
+  }
+
+  onGeometryStyleChange(style) {
+    this.geometryStyle = style;
```

Comment:
I believe we can remove the need for this logic if we called `this.props.onChange('style', this.outputGeometry)` instead of `this.props.onGeometryStyleChange(style)` at the end of the `generateOutputGeometry` method in our MapDrawingToolbar component.

Alternatively, I think we could also remove this logic by refactoring the `onFieldChange` method in PlaceDetailEditor to update changes for multiple attributes. Then we can call something like `this.props.onChange({ [this.props.name]: this.outputGeometry, style: style })` in our MapDrawingToolbar component, and similarly remove the `onGeometryStyleChange` method.

Reply:
So in the second idea above, how would `onFieldChange` handle an object like: `{ [this.props.name]: this.outputGeometry, style: style }`? It seems like we'd need to give it logic to split apart the special `style` property and whatever the main field value is, or we'd need to pass the name of the field as a separate argument.

Reply:
Are you talking about how to set the `updatingField` value in the `onFieldChange` method? Yeah, I think I missed that detail - so maybe you'd have to pass the "main" field value in separately.

FWIW having to track `updatingField` is a bit of a code smell. I'm not sure how to fix that though :thinking:

Reply:
Right-- how to set the `updatingField`, but also how to set the field value itself in this line:
```
fields: fields.set(fieldName, fieldStatus),
```
Yeah the `updatingField` always felt like a hack. Its only purpose is so we can prevent unnecessary rendering of `FormField`s as the user interacts with the form.

Reply:
In essence, you can do this:

```
fields: fields.set(fieldName1, fieldStatus1).set(fieldName2, fieldStatus2).set(fieldName3, fieldStatus3), // etc
```
where you can get all the `fieldName`'s and `fieldStatus`'s from the object of mappings that you pass in. More specifically, the implementation would be something like this:

```
onFieldChange({ fieldMappings, mainField, isInitializing }) {
  this.setState(({ fields }) => {
    Object.keys(fieldMappings).forEach(fieldName => {
      fields = fields.set(fieldName, fieldMappings[fieldName])
    })
    return {
      fields: fields,
      updatingField: mainField,
      isInitializing: isInitializing,
    }
  });
}

```

Does that make sense? Let me know if I'm missing something.

Reply:
Right, I see what you're getting at. The other issue, though, is that the `MapDrawingToolbar` is the only "field" that returns two values: the geometry itself and the geometry style object. Creating a piece of state for both of these values is problematic, because for various reasons (like validation) we expect each field to return a single value.

That's why we have a separate `onGeometryStyleChange` method, and why we tack the style object on at the end, right before submission.

My feeling is that the real issue here lies in the fact that the `MapDrawingToolbar` is not really a field, and we're trying to cram it into the field abstraction. Down the road I feel we should come up with a "toolbar" abstraction or something like that.

For now, though, I think it would be easier to keep `onGeometryStyleChange` in place. What do you think?